src/components/SlateEditor/plugins/toolbar/toolbarState.ts

```diff
@@ -238,95 +236,70 @@ export const createToolbarDefaultValues = (userValues: CategoryFilters = {}): Ca
   }, {});
 };
 
-interface ToolbarStateProps {
-  options?: CategoryFilters;
-  areaOptions?: AreaFilters;
-  editorAncestors?: Element[];
-  selection?: BaseSelection;
-}
-
-export const getSelectionElements = (editor: Editor, selection: BaseSelection): Element[] => {
-  // Get elements in the selection only
-  if (selection) {
-    const fragments = editor.getFragment();
-    const elementFragments = fragments.filter(Element.isElement);
-
-    // When a fragment selects multiple blocks it seems to return the blocks in a section,
-    // if it is a section we have to flatten it and find the inline children.
-    // TODO: This logic works for the current nodes we have, further nested might provide a problem and must be solved in a different way.
-    if (elementFragments?.[0]?.type === TYPE_SECTION) {
-      return elementFragments
-        .flatMap((node) => node?.children as Element[])
-        .flatMap((node) => node?.children as Element[]);
-    } else if (elementFragments?.[0]?.type === TYPE_PARAGRAPH) {
-      return elementFragments.flatMap((node) => node?.children as Element[]);
-    }
-
-    return elementFragments;
-  }
-  return [];
+type SelectionElementTypes = {
+  elementTypes: ElementType[] | undefined;
+  multipleBlocksSelected: boolean;
 };
 
-export const getEditorAncestors = (editor: Editor, reverse?: boolean): Element[] => {
-  // Finds the current lowest node in the editor and creates an array of its ancestors.
-  const [lowest] = Editor.nodes(editor, { mode: "lowest" });
-  const ancestorGen = Node.ancestors(editor, lowest[1], { reverse });
-  const elementAncestors: Element[] = [];
-  // ancestorGen is a generator, so we need to iterate over it to get the values.
-  for (const ancestor of ancestorGen) {
-    if (Element.isElement(ancestor[0]) && ancestor[0].type !== "section") {
-      elementAncestors.push(ancestor[0]);
+export const selectionElementTypes = (editor: Editor, rawSelection: Selection): SelectionElementTypes => {
```

Comment:
Tjener vi egentlig noe på å returnere kun type her? Er det dumt dersom vi plutselig har behov for noden?